modules/users/server/controllers/users.profile.server.controller.js

```diff
@@ -854,19 +890,18 @@ exports.userByUsername = function (req, res, next, username) {
               });
             }
 
-            const isOwnProfile = req.user._id.equals(profile._id);
-            const isBannedProfile =
-              profile.roles.includes('suspended') ||
-              profile.roles.includes('shadowban');
-            const isAdminOrModerator =
-              req.user.roles.includes('moderator') ||
-              req.user.roles.includes('admin');
+            const {
+              isAdminOrModerator,
+              isOwnProfile,
+              isBannedProfile,
+              isBlocked,
+            } = classifyPermission(req.user, profile);
 
             // Not own profile, and not public, or suspended, or shadowbanned user
             if (
               !isAdminOrModerator &&
               !isOwnProfile &&
-              (!profile.public || isBannedProfile)
+              (!profile.public || isBannedProfile || isBlocked)
```

Comment:
Definitely not a major thing, but maybe something to think of in the future. It took me some time to understand the logic whether a user should be visible or not (I started working with the codebase a few days ago). I wonder if it would make sense to refactor/rename/simplify things here and also in the `userMiniByID` function. I could imagine this could be an easy place for a bug to hide:

- `user` vs `profile` - who is me, who is the other user
- `isAdminOrModerator`, `isBannedProfile` - me or the other user?
- the comment above doesn't help much especially because it can get outdated easily